src/frontend/src/components/dappList.ts

```diff
@@ -0,0 +1,119 @@
+import { html, TemplateResult } from "lit-html";
+
+export type DappDescription = {
+  id: string;
+  name: string;
+  oneLiner?: string;
+  link: string;
+  derivationOrigin?: string;
+  description: string;
+  logo: string;
+};
+
+const createMarqueeListRowItems = (
+  dapps: DappDescription[]
+): TemplateResult => {
+  return html` ${dapps.map((dapp) => {
+    const { logo, name } = dapp;
+
+    return html`<div class="c-marquee__item">
+      <img
+        src="icons/${logo}"
+        alt="${name}"
+        class="c-marquee__image"
+        loading="lazy"
+      />
+    </div>`;
+  })}`;
+};
+
+/**
+ * The marquee is a list of dapps that scrolls horizontally.
+ * It uses a CSS keyframe animation to scroll the each row alternating from left to right and viceversa.
+ * To give the illusion of an infinite scrolling, the list is duplicated horizontally.
+ * when the animation reaches the end of the list, it is reset to the beginning. (default keyframe animation behaviour)
+ */
+
+const createMarqueeList = (dapps: DappDescription[]): TemplateResult => {
+  const itemsPerRow = 5;
+  const totalRows = 4;
+
+  const rows = [];
+
+  for (let i = 0; i < totalRows; i++) {
+    const row = [];
+    for (let j = 0; j < itemsPerRow; j++) {
+      const dapp = dapps[(i * itemsPerRow + j) % dapps.length];
+      row.push(dapp);
+    }
+    rows.push([...row]);
+  }
```

Comment:
this seems a bit error prone. How about splitting the array into chunks instead? https://stackoverflow.com/a/8495740

Reply:
Ive updated my code, please have a look and tell me if you understand my implementation now if it works for you